__tests__/app.test.js

```diff
@@ -61,14 +61,91 @@ describe("GET /api/reviews", () => {
       .get("/api/reviews")
       .expect(200)
       .then((result) => {
         expect(result.body.reviews).toBeSortedBy("created_at", { descending: true });
       });
   });
+  describe("Queries", () => {
+    describe("Sort_By", () => {
+      test("200: should sort by given column", () => {
+        return request(app)
+          .get("/api/reviews?sort_by=title")
+          .expect(200)
+          .then((result) => {
+            expect(result.body.reviews).toBeSortedBy("title", { descending: true });
+          });
+      });
+      test("400: should reject columns that arent in the DB", () => {
+        return request(app)
+          .get("/api/reviews?sort_by=randomcolumnname")
+          .expect(400)
+          .then((result) => {
+            expect(result.body.msg).toBe("Bad Request");
+          });
+      });
+    });
+    describe("Order", () => {
+      test("200: should order by ascending", () => {
+        return request(app)
+          .get("/api/reviews?order=asc")
+          .expect(200)
+          .then((result) => {
+            expect(result.body.reviews).toBeSortedBy("created_at");
+          });
+      });
+      test("200: should order by descending", () => {
+        return request(app)
+          .get("/api/reviews?order=desc")
+          .expect(200)
+          .then((result) => {
+            expect(result.body.reviews).toBeSortedBy("created_at", { descending: true });
+          });
+      });
+      test("200: should default to descending given an invalid argument", () => {
+        return request(app)
+          .get("/api/reviews?order=AnyOrderPlease")
+          .expect(200)
+          .then((result) => {
+            expect(result.body.reviews).toBeSortedBy("created_at", { descending: true });
+          });
+      });
+    });
+    describe("Category", () => {
+      test("200: should return reviews with a given category", () => {
+        return request(app)
+          .get("/api/reviews?category=euro game")
+          .expect(200)
+          .then((result) => {
+            expect(result.length > 0);
+            result.body.reviews.forEach((category) => {
+              expect(category).toMatchObject({
+                owner: expect.any(String),
+                title: expect.any(String),
+                review_id: expect.any(Number),
+                category: "euro game",
+                review_img_url: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                designer: expect.any(String),
+                comment_count: expect.any(String),
+              });
+            });
+          });
+      });
+      test("404: should reject category for invalid category name", () => {
+        return request(app)
+          .get("/api/reviews?category=silly game")
+          .expect(404)
+          .then((result) => {
+            expect(result.body.msg).toBe("Resource not found");
+          });
+      });
+    });
+  });
```

Comment:
Good test coverage! But what about the case where the category _does_ exist, it just doesn't have any reviews in it?